toplevel/auth/auth.go

```diff
@@ -143,16 +137,36 @@ func (c config) Apply(entriesBytes []byte, dryRun bool) {
 			existingPolicyMappings := make([]policyMapping, 0)
 			entitiesList := vault.ListSecrets(filepath.Join("/auth", e.Path, "map/teams"))
 			if entitiesList != nil {
-				for _, entity := range entitiesList.Data["keys"].([]interface{}) {
-					policyMappingPath := filepath.Join("/auth/", e.Path, "map/teams", entity.(string))
-					policiesMappedToEntity := vault.ReadSecret(policyMappingPath).Data["value"].(string)
-					policies := make([]map[string]interface{}, 0)
-					for _, policy := range strings.Split(policiesMappedToEntity, ",") {
-						policies = append(policies, map[string]interface{}{"name": policy})
-					}
-					existingPolicyMappings = append(existingPolicyMappings,
-						policyMapping{GithubTeam: map[string]interface{}{"team": entity}, Policies: policies})
+
+				var mutex = &sync.Mutex{}
+
+				entities := entitiesList.Data["keys"].([]interface{})
+
+				bwg := utils.NewBoundedWaitGroup(threadPoolSize)
+
+				// fill existing policy mappings array in parallel
+				for i := range entities {
+
+					bwg.Add(1)
+
+					go func(i int) {
+
+						mutex.Lock()
+
+						policyMappingPath := filepath.Join("/auth/", e.Path, "map/teams", entities[i].(string))
+						policiesMappedToEntity := vault.ReadSecret(policyMappingPath).Data["value"].(string)
+						policies := make([]map[string]interface{}, 0)
+						for _, policy := range strings.Split(policiesMappedToEntity, ",") {
+							policies = append(policies, map[string]interface{}{"name": policy})
+						}
+						existingPolicyMappings = append(existingPolicyMappings,
```

Comment:
I would lock just before this line, as this is the only one accessing a shared resource (one that is defined outside the scope of the go routine)

Reply:
totally!

Reply:
changes pushed!